eth/catalyst/celo_simulated_beacon.go

```diff
@@ -0,0 +1,49 @@
+package catalyst
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+const (
+	// PostEcotoneGasParamsLength is the length of gas parameters after the Ecotone fork
+	// See - types.extractL1GasParamsPostEcotone
+	PostEcotoneGasParamsLength = 164
+)
+
+func (c *SimulatedBeacon) payloadGasLimit() *uint64 {
+	if c.eth.BlockChain().Config().Optimism == nil {
+		return nil
+	}
+	// If Optimism config is set we need to set the gas limit in the payload attributes.
+	return &c.eth.BlockChain().CurrentBlock().GasLimit
+}
+
+func (c *SimulatedBeacon) payloadSystemTransaction() ([][]byte, error) {
+	// Post ecotone we need to provide a system transaction to set L1 gas params, we don't actually need to
+	// set realistic values for the gas params, since in Celo we ensure L1 gas fees of zero.
+	// However this is required in order to be able to correctly derive receipt fields.
+	// See types.Receipts.DeriveFields.
```

Comment:
```suggestion
	// Post ecotone we need to provide a system transaction to set L1 gas params.
	// This mechanism doesn't work in the STF, since no L2 contracts are pre-deployed.
	// The l1 gas-parameters are defaulting to 0, which is coincidentally what we expect in Celo.
	// This is why we don't need to set realistic values in the system transaction here, except for 
	// the `Data` field, which is used by `types.Receipts.DeriveFields` and expected to be all 0 to match 
	// what is deducted in the STF.
```